src/static/js/ace2_inner.js

```diff
@@ -2922,6 +2930,36 @@ function Ace2Inner(){
     //console.log("%o %o %s", rep.selStart, rep.selEnd, rep.selFocusAtStart);
   }
 
+  function ifAttribIsFormattingUpdateStyleButtonState(attribName, hasStyleOnRepSelection) {
+    if (attributeIsFormattingStyle(attribName)) {
```

Comment:
this checking is duplicated when `ifAttribIsFormattingUpdateStyleButtonState()` is called from `selectFormattingButtonIfLineHasStyleApplied` -- we are looping through the `FORMATTING_STYLES`, so `attribName` surely is a formatting style. To avoid this, you could turn `ifAttribIsFormattingUpdateStyleButtonState` into a function that simply toggles the class on the button, and extract the checking (`if (attributeIsFormattingStyle(attribName))`) to the other place where it is called -- on `toggleAttributeOnSelection`.